In [1]:
import pandas as pd
import numpy as np

In [2]:
# 연결데이터 , 개별데이터 불러오기
df_con = pd.read_csv("./데이터자료/1. 상장폐지종목 정리끝.csv",index_col=0)
df_invis = pd.read_csv("./데이터자료/1. 코스닥 기업 개별재무제표.csv",encoding='cp949')

In [3]:
df_con.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10450 entries, 0 to 10449
Data columns (total 60 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   회사명                       10450 non-null  object 
 1   거래소코드                     10450 non-null  int64  
 2   회계년도                      10450 non-null  object 
 3   상장일                       10450 non-null  object 
 4   상장폐지일                     860 non-null    object 
 5   소속코드                      10450 non-null  int64  
 6   산업명                       10450 non-null  object 
 7   [제조]총자본증가율(IFRS연결)        7906 non-null   float64
 8   [제조]유형자산증가율(IFRS연결)       7906 non-null   float64
 9   [제조]비유동자산증가율(IFRS연결)      7906 non-null   float64
 10  [제조]유동자산증가율(IFRS연결)       7906 non-null   float64
 11  [제조]재고자산증가율(IFRS연결)       7906 non-null   float64
 12  [제조]자기자본증가율(IFRS연결)       7906 non-null   float64
 13  [제조]매출액증가율(IFRS연결)        7906 non-null   float64
 14  [제조]순이

In [4]:
df_con[:20]

회사명  거래소코드        회계년도         상장일 상장폐지일  소속코드                 산업명  \
0   (주)CMG제약  58820  2011-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
1   (주)CMG제약  58820  2012-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
2   (주)CMG제약  58820  2013-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
3   (주)CMG제약  58820  2014-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
4   (주)CMG제약  58820  2015-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
5   (주)CMG제약  58820  2016-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
6   (주)CMG제약  58820  2017-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
7   (주)CMG제약  58820  2018-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
8   (주)CMG제약  58820  2019-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
9   (주)CMG제약  58820  2020-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
10   (주)ES큐브  50120  2011-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
11   (주)ES큐브  50120  2012-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
12   (주)ES큐브  50120  2013-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
13   (주)ES큐브  50120  2014-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
14   (주)ES큐브  50120  2015-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
15   (주)ES큐브  50120  2016-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
16   (주)ES큐브  50120  2017-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
17   (주)ES큐브  50120  2018-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
18   (주)ES큐브  50120  2019-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
19   (주)ES큐브  50120  2020-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   

    [제조]총자본증가율(IFRS연결)  [제조]유형자산증가율(IFRS연결)  [제조]비유동자산증가율(IFRS연결)  ...  \
0                 1.98                -5.90                -12.89  ...   
1               109.15                -4.12                -16.43  ...   
2                -2.09                16.09                 77.05  ...   
3                 3.86                74.30                 68.15  ...   
4                -2.64                -5.37                  1.66  ...   
5                96.31                -5.14                  6.43  ...   
6                -1.27                -4.28                -12.01  ...   
7                 4.39                 6.71                 16.47  ...   
8                68.07                39.67                 26.20  ...   
9                  NaN                  NaN                   NaN  ...   
10                0.00                 0.00                  0.00  ...   
11                0.00                 0.00                  0.00  ...   
12                1.85               -31.56                -62.76  ...   
13               20.10                 5.59                 18.41  ...   
14               -1.75               -14.40                 38.88  ...   
15               89.08               258.91                275.90  ...   
16                0.69                64.55                  1.06  ...   
17                6.17                30.79                 22.82  ...   
18              312.66                14.02                 55.75  ...   
19               54.08                69.21                274.24  ...   

    [제조]기계투자효율(IFRS연결)  [제조]부가가치율(IFRS연결)  [제조]노동소득분배율(IFRS연결)  \
0               128.51              19.56                89.82   
1                10.67               1.66              1749.87   
2               289.40              34.29                71.28   
3               132.06              31.04                79.30   
4                90.92              15.96               144.10   
5               273.60              31.90                63.53   
6               389.08              33.01                64.09   
7               478.91              33.79                67.33   
8               438.05              32.47                72.60   
9                  NaN                NaN                  NaN   
10             1125.26              22.56               120.20   
11

In [5]:
df_invis.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14347 entries, 0 to 14346
Data columns (total 55 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   회사명                     14347 non-null  object 
 1   거래소코드                   14347 non-null  int64  
 2   회계년도                    14347 non-null  object 
 3   상장일                     14347 non-null  object 
 4   상장폐지일                   1534 non-null   object 
 5   소속코드                    14347 non-null  int64  
 6   산업명                     14347 non-null  object 
 7   [제조]총자본증가율(IFRS)        12300 non-null  float64
 8   [제조]유형자산증가율(IFRS)       12300 non-null  float64
 9   [제조]비유동자산증가율(IFRS)      12300 non-null  float64
 10  [제조]유동자산증가율(IFRS)       12300 non-null  float64
 11  [제조]재고자산증가율(IFRS)       12300 non-null  float64
 12  [제조]자기자본증가율(IFRS)       12300 non-null  float64
 13  [제조]매출액증가율(IFRS)        12300 non-null  float64
 14  [제조]순이익증가율(IFRS)        12300 non-null

In [6]:
df_invis[:20]

회사명  거래소코드     회계년도         상장일 상장폐지일  소속코드                 산업명  \
0   (주)CMG제약  58820  2011/12  2001/08/31   NaN     5    의료용 물질 및 의약품 제조업   
1   (주)CMG제약  58820  2012/12  2001/08/31   NaN     5    의료용 물질 및 의약품 제조업   
2   (주)CMG제약  58820  2013/12  2001/08/31   NaN     5    의료용 물질 및 의약품 제조업   
3   (주)CMG제약  58820  2014/12  2001/08/31   NaN     5    의료용 물질 및 의약품 제조업   
4   (주)CMG제약  58820  2015/12  2001/08/31   NaN     5    의료용 물질 및 의약품 제조업   
5   (주)CMG제약  58820  2016/12  2001/08/31   NaN     5    의료용 물질 및 의약품 제조업   
6   (주)CMG제약  58820  2017/12  2001/08/31   NaN     5    의료용 물질 및 의약품 제조업   
7   (주)CMG제약  58820  2018/12  2001/08/31   NaN     5    의료용 물질 및 의약품 제조업   
8   (주)CMG제약  58820  2019/12  2001/08/31   NaN     5    의료용 물질 및 의약품 제조업   
9   (주)CMG제약  58820  2020/12  2001/08/31   NaN     5    의료용 물질 및 의약품 제조업   
10   (주)ES큐브  50120  2011/12  2002/04/25   NaN     5         도매 및 상품 중개업   
11   (주)ES큐브  50120  2012/12  2002/04/25   NaN     5         도매 및 상품 중개업   
12   (주)ES큐브  50120  2013/12  2002/04/25   NaN     5         도매 및 상품 중개업   
13   (주)ES큐브  50120  2014/12  2002/04/25   NaN     5         도매 및 상품 중개업   
14   (주)ES큐브  50120  2015/12  2002/04/25   NaN     5         도매 및 상품 중개업   
15   (주)ES큐브  50120  2016/12  2002/04/25   NaN     5         도매 및 상품 중개업   
16   (주)ES큐브  50120  2017/12  2002/04/25   NaN     5         도매 및 상품 중개업   
17   (주)ES큐브  50120  2018/12  2002/04/25   NaN     5         도매 및 상품 중개업   
18   (주)ES큐브  50120  2019/12  2002/04/25   NaN     5         도매 및 상품 중개업   
19   (주)ES큐브  50120  2020/12  2002/04/25   NaN     5         도매 및 상품 중개업   

    [제조]총자본증가율(IFRS)  [제조]유형자산증가율(IFRS)  [제조]비유동자산증가율(IFRS)  ...  \
0               2.89              -6.44              -10.56  ...   
1             106.87              -4.02              -19.28  ...   
2              -1.80              16.63               78.54  ...   
3               3.87              74.30               68.15  ...   
4              -2.64              -5.37                1.66  ...   
5              96.11              -5.14                6.43  ...   
6              -1.27              -4.28              -12.01  ...   
7               4.39               6.71               16.47  ...   
8              68.07              39.67               26.20  ...   
9               1.62              23.31               18.52  ...   
10             -5.43             -11.82               22.79  ...   
11             -1.23              -0.80                9.82  ...   
12              6.34             -34.47              -48.39  ...   
13             -5.46             -30.00               56.34  ...   
14              4.92              -6.88                4.68  ...   
15            121.67             683.64              148.93  ...   
16              0.37              60.62               -1.61  ...   
17              3.46              44.96               32.18  ...   
18             25.43              -2.80               40.72  ...   
19            -39.74             -94.76              -33.37  ...   

    [제조]유동자산회전률(IFRS)  [제조]당좌자산회전률(IFRS)  [제조]부가가치(IFRS)  [제조]총자본투자효율(IFRS)  \
0                1.25               1.56         3669.25              13.32   
1                0.52               0.57          -88.04              -0.15   
2                0.48               0.52         6952.08              12.42   
3                0.68               0.80         7065.04              12.15   
4                0.99               1.22         4316.80               7.63   
5                0.63               0.71        10518.86               9.48   
6                0.52               0.57        13710.01              12.51   
7                0.61               0.68        16847.65              14.73   
8                0.51               0.55        19033.25               9.90   
9                0.43               0.47        17019.70               8.71   
10               1.85               1.90         7961.63              17.60   
11               2.57              

In [7]:
#컬럼명 일치를 위한 컬럼명 변경 
df_con.columns = df_con.columns.str.replace('연결','')
df_con.columns

Index(['회사명', '거래소코드', '회계년도', '상장일', '상장폐지일', '소속코드', '산업명',
       '[제조]총자본증가율(IFRS)', '[제조]유형자산증가율(IFRS)', '[제조]비유동자산증가율(IFRS)',
       '[제조]유동자산증가율(IFRS)', '[제조]재고자산증가율(IFRS)', '[제조]자기자본증가율(IFRS)',
       '[제조]매출액증가율(IFRS)', '[제조]순이익증가율(IFRS)', '[제조]매출액총이익률(IFRS)',
       '[제조]매출액순이익률(IFRS)', '[제조]자기자본순이익률(IFRS)', '[제조]총자본사업이익률(IFRS)',
       '[제조]총자본순이익률(IFRS)', '[제조]1주당매출액(IFRS)', '[제조]1주당순이익(IFRS)',
       '[제조]유동비율(IFRS)', '[제조]당좌비율(IFRS)', '[제조]현금비율(IFRS)',
       '[제조]순운전자본비율(IFRS)', '[제조]부채비율(IFRS)', '[제조]이자보상배율(이자비용)(IFRS)',
       '[제조]비유동비율(IFRS)', '[제조]유동자산구성비율(IFRS)', '[제조]자기자본구성비율(IFRS)',
       '[제조]자기자본배율(IFRS)', '[제조]총자본회전률(IFRS)', '[제조]자기자본회전률(IFRS)',
       '[제조]비유동자산회전률(IFRS)', '[제조]재고자산회전률(IFRS)', '[제조]매출채권회전률(IFRS)',
       '[제조]매입채무회전률(IFRS)', '[제조]매입채무회전기간(IFRS)', '[제조]재고자산회전기간(IFRS)',
       '[제조]매출채권회전기간(IFRS)', '[제조]1회전기간(IFRS)', '[제조]운전자본회전률(IFRS)',
       '[제조]순운전자본회전률(IFRS)', '[제조]유형자산회전율(IFRS)', '[제조]유동자산회전률(IFRS)',
       '[제조]당좌자산회전률(IFRS)', '[제조]부가가

In [8]:
# 빈 값을 넣기 위한 list 생성
list = ['[제조]총자본증가율(IFRS)', '[제조]유형자산증가율(IFRS)', '[제조]비유동자산증가율(IFRS)',
       '[제조]유동자산증가율(IFRS)', '[제조]재고자산증가율(IFRS)', '[제조]자기자본증가율(IFRS)',
       '[제조]매출액증가율(IFRS)', '[제조]순이익증가율(IFRS)', '[제조]매출액총이익률(IFRS)',
       '[제조]매출액순이익률(IFRS)', '[제조]자기자본순이익률(IFRS)', '[제조]총자본사업이익률(IFRS)',
       '[제조]총자본순이익률(IFRS)', '[제조]1주당매출액(IFRS)', '[제조]1주당순이익(IFRS)',
       '[제조]유동비율(IFRS)', '[제조]당좌비율(IFRS)', '[제조]현금비율(IFRS)',
       '[제조]순운전자본비율(IFRS)', '[제조]부채비율(IFRS)', '[제조]이자보상배율(이자비용)(IFRS)',
       '[제조]비유동비율(IFRS)', '[제조]유동자산구성비율(IFRS)', '[제조]자기자본구성비율(IFRS)',
       '[제조]자기자본배율(IFRS)', '[제조]총자본회전률(IFRS)', '[제조]자기자본회전률(IFRS)',
       '[제조]비유동자산회전률(IFRS)', '[제조]재고자산회전률(IFRS)', '[제조]매출채권회전률(IFRS)',
       '[제조]매입채무회전률(IFRS)', '[제조]매입채무회전기간(IFRS)', '[제조]재고자산회전기간(IFRS)',
       '[제조]매출채권회전기간(IFRS)', '[제조]1회전기간(IFRS)', '[제조]운전자본회전률(IFRS)',
       '[제조]순운전자본회전률(IFRS)', '[제조]유형자산회전율(IFRS)', '[제조]유동자산회전률(IFRS)',
       '[제조]당좌자산회전률(IFRS)', '[제조]부가가치(IFRS)', '[제조]총자본투자효율(IFRS)',
       '[제조]설비투자효율(IFRS)', '[제조]기계투자효율(IFRS)', '[제조]부가가치율(IFRS)',
       '[제조]노동소득분배율(IFRS)', '[제조]자본분배율(IFRS)', '[제조]이윤분배율(IFRS)']

In [9]:
# 빈공간을 개별로 넣기 
for i in list:
    df_con[i] = df_con[i].fillna(df_invis[i])
    print(df_con[i])
    

0          1.98
1        109.15
2         -2.09
3          3.86
4         -2.64
          ...  
10445     -2.26
10446      2.71
10447      2.47
10448     -3.60
10449      1.77
Name: [제조]총자본증가율(IFRS), Length: 10450, dtype: float64
0        -5.90
1        -4.12
2        16.09
3        74.30
4        -5.37
         ...  
10445    -2.56
10446    -7.36
10447    -2.78
10448    -0.50
10449    11.95
Name: [제조]유형자산증가율(IFRS), Length: 10450, dtype: float64
0       -12.89
1       -16.43
2        77.05
3        68.15
4         1.66
         ...  
10445    -1.88
10446    37.82
10447    -3.98
10448    10.54
10449    25.00
Name: [제조]비유동자산증가율(IFRS), Length: 10450, dtype: float64
0         17.64
1        207.08
2        -18.89
3        -25.93
4         -7.17
          ...  
10445     -2.34
10446     -4.96
10447      4.51
10448     -7.71
10449     -6.33
Name: [제조]유동자산증가율(IFRS), Length: 10450, dtype: float64
0         -1.43
1          7.73
2         56.29
3          6.37
4         11.04
          ...  
10

In [10]:
df_con.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10450 entries, 0 to 10449
Data columns (total 60 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   회사명                     10450 non-null  object 
 1   거래소코드                   10450 non-null  int64  
 2   회계년도                    10450 non-null  object 
 3   상장일                     10450 non-null  object 
 4   상장폐지일                   860 non-null    object 
 5   소속코드                    10450 non-null  int64  
 6   산업명                     10450 non-null  object 
 7   [제조]총자본증가율(IFRS)        10061 non-null  float64
 8   [제조]유형자산증가율(IFRS)       10061 non-null  float64
 9   [제조]비유동자산증가율(IFRS)      10061 non-null  float64
 10  [제조]유동자산증가율(IFRS)       10061 non-null  float64
 11  [제조]재고자산증가율(IFRS)       10061 non-null  float64
 12  [제조]자기자본증가율(IFRS)       10061 non-null  float64
 13  [제조]매출액증가율(IFRS)        10061 non-null  float64
 14  [제조]순이익증가율(IFRS)        10061 non-null

In [11]:
df_con[:20]

회사명  거래소코드        회계년도         상장일 상장폐지일  소속코드                 산업명  \
0   (주)CMG제약  58820  2011-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
1   (주)CMG제약  58820  2012-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
2   (주)CMG제약  58820  2013-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
3   (주)CMG제약  58820  2014-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
4   (주)CMG제약  58820  2015-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
5   (주)CMG제약  58820  2016-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
6   (주)CMG제약  58820  2017-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
7   (주)CMG제약  58820  2018-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
8   (주)CMG제약  58820  2019-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
9   (주)CMG제약  58820  2020-12-01  2001-08-31   NaN     5    의료용 물질 및 의약품 제조업   
10   (주)ES큐브  50120  2011-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
11   (주)ES큐브  50120  2012-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
12   (주)ES큐브  50120  2013-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
13   (주)ES큐브  50120  2014-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
14   (주)ES큐브  50120  2015-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
15   (주)ES큐브  50120  2016-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
16   (주)ES큐브  50120  2017-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
17   (주)ES큐브  50120  2018-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
18   (주)ES큐브  50120  2019-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   
19   (주)ES큐브  50120  2020-12-01  2002-04-25   NaN     5         도매 및 상품 중개업   

    [제조]총자본증가율(IFRS)  [제조]유형자산증가율(IFRS)  [제조]비유동자산증가율(IFRS)  ...  \
0               1.98              -5.90              -12.89  ...   
1             109.15              -4.12              -16.43  ...   
2              -2.09              16.09               77.05  ...   
3               3.86              74.30               68.15  ...   
4              -2.64              -5.37                1.66  ...   
5              96.31              -5.14                6.43  ...   
6              -1.27              -4.28              -12.01  ...   
7               4.39               6.71               16.47  ...   
8              68.07              39.67               26.20  ...   
9               1.62              23.31               18.52  ...   
10              0.00               0.00                0.00  ...   
11              0.00               0.00                0.00  ...   
12              1.85             -31.56              -62.76  ...   
13             20.10               5.59               18.41  ...   
14             -1.75             -14.40               38.88  ...   
15             89.08             258.91              275.90  ...   
16              0.69              64.55                1.06  ...   
17              6.17              30.79               22.82  ...   
18            312.66              14.02               55.75  ...   
19             54.08              69.21              274.24  ...   

    [제조]기계투자효율(IFRS)  [제조]부가가치율(IFRS)  [제조]노동소득분배율(IFRS)  [제조]자본분배율(IFRS)  \
0             128.51            19.56              89.82            10.18   
1              10.67             1.66            1749.87         -1649.87   
2             289.40            34.29              71.28            28.72   
3             132.06            31.04              79.30            20.70   
4              90.92            15.96             144.10           -44.10   
5             273.60            31.90              63.53            36.47   
6             389.08            33.01              64.09            35.91   
7             478.91            33.79              67.33            32.67   
8             438.05            32.47              72.60            27.40   
9             475.13            26.40              86.14            13.86   
10           1125.26            22.56             120.20           -20.20

In [12]:
# 개별재무제표가 없는 데이터의 인덱스를 추출해서 없애기
a= df_con[df_con["[제조]총자본증가율(IFRS)"].isnull()].index
df_con.drop(a,inplace=True)

In [13]:
df_con.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10061 entries, 0 to 10449
Data columns (total 60 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   회사명                     10061 non-null  object 
 1   거래소코드                   10061 non-null  int64  
 2   회계년도                    10061 non-null  object 
 3   상장일                     10061 non-null  object 
 4   상장폐지일                   828 non-null    object 
 5   소속코드                    10061 non-null  int64  
 6   산업명                     10061 non-null  object 
 7   [제조]총자본증가율(IFRS)        10061 non-null  float64
 8   [제조]유형자산증가율(IFRS)       10061 non-null  float64
 9   [제조]비유동자산증가율(IFRS)      10061 non-null  float64
 10  [제조]유동자산증가율(IFRS)       10061 non-null  float64
 11  [제조]재고자산증가율(IFRS)       10061 non-null  float64
 12  [제조]자기자본증가율(IFRS)       10061 non-null  float64
 13  [제조]매출액증가율(IFRS)        10061 non-null  float64
 14  [제조]순이익증가율(IFRS)        10061 non-null

In [14]:
a = df_con.columns[9]
a

'[제조]비유동자산증가율(IFRS)'

In [15]:
a[4:]

'비유동자산증가율(IFRS)'

In [16]:
# 추가 다른 컬럼에서 결측치가 있는 데이터 삭제 
b  = df_con[df_con["[제조]이윤분배율(IFRS)"].isnull()].index
df_con.drop(b, inplace=True)

In [17]:
# 인덱스 초기화
df_con.reset_index(drop='index', inplace=True)

In [18]:
# 상장폐지일이 존재하면서 소속코드가 5인 데이터들의 상장폐지 일 내용 삭제
a = df_con[~df_con["상장폐지일"].isnull()]
b = a[a["소속코드"] ==5].index
b

Int64Index([ 661,  662, 2201, 2202, 2203, 2204, 2205, 2206, 3790, 3791, 3792,
            3793, 3794, 3795, 3796, 3797, 5455, 5456, 5457, 5458, 5459, 5460,
            6596, 6597, 6598, 6599, 6600, 6601, 6602, 6603, 8601, 8602, 8603,
            8604, 8605, 8606, 8607, 8608, 8609, 9694, 9695, 9696, 9697, 9698,
            9699],
           dtype='int64')

In [19]:
# replace로 대체
for i in b:
    df_con["상장폐지일"] = df_con["상장폐지일"].replace(df_con.loc[i, "상장폐지일"], np.NAN)

In [20]:
# 대체 확인
c = df_con[~df_con["상장폐지일"].isnull()]
len(c[c["소속코드"] ==5]["회사명"].unique())

0

In [21]:
print(len(df_con["회사명"].unique()))
print(len(df_con[df_con["소속코드"]==5]["회사명"].unique()))
print(len(df_con[df_con["소속코드"]==4]["회사명"].unique()))

1439
1204
178


## 총데이터 10061
## 총 기업수 1439 정상기업 1261 부도기업 178
## 우리의 부도기업 정의 에 따라 정상데이터 9083 부도데이터 178

### 컬럼이름 간단하게 하기위해 수정
### 상장폐지 사유 컬럼 삭제

In [22]:
df_con.columns = df_con.columns.str.replace("제조",'')
df_con.columns = df_con.columns.str.replace('[','')
df_con.columns = df_con.columns.str.replace(']','')
df_con.columns = df_con.columns.str.replace('IFRS','')
df_con.columns = df_con.columns.str.replace("연결",'')
df_con.columns = df_con.columns.str.replace("(",'')
df_con.columns = df_con.columns.str.replace(")",'')

C:\Users\young\AppData\Local\Temp\ipykernel_4344\3807298736.py:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df_con.columns = df_con.columns.str.replace('[','')
C:\Users\young\AppData\Local\Temp\ipykernel_4344\3807298736.py:3: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df_con.columns = df_con.columns.str.replace(']','')
C:\Users\young\AppData\Local\Temp\ipykernel_4344\3807298736.py:6: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df_con.columns = df_con.columns.str.replace("(",'')
C:\Users\young\AppData\Local\Temp\

In [23]:
df_con.drop('상장폐지사유',axis=1,inplace=True)

In [24]:
df_con

회사명  거래소코드        회계년도         상장일 상장폐지일  소속코드  \
0      (주)CMG제약  58820  2011-12-01  2001-08-31   NaN     5   
1      (주)CMG제약  58820  2012-12-01  2001-08-31   NaN     5   
2      (주)CMG제약  58820  2013-12-01  2001-08-31   NaN     5   
3      (주)CMG제약  58820  2014-12-01  2001-08-31   NaN     5   
4      (주)CMG제약  58820  2015-12-01  2001-08-31   NaN     5   
...         ...    ...         ...         ...   ...   ...   
10056   흥구석유(주)  24060  2016-12-01  1994-12-07   NaN     5   
10057   흥구석유(주)  24060  2017-12-01  1994-12-07   NaN     5   
10058   흥구석유(주)  24060  2018-12-01  1994-12-07   NaN     5   
10059   흥구석유(주)  24060  2019-12-01  1994-12-07   NaN     5   
10060   흥구석유(주)  24060  2020-12-01  1994-12-07   NaN     5   

                      산업명  총자본증가율  유형자산증가율  비유동자산증가율  ...  설비투자효율   기계투자효율  \
0        의료용 물질 및 의약품 제조업    1.98    -5.90    -12.89  ...   35.86   128.51   
1        의료용 물질 및 의약품 제조업  109.15    -4.12    -16.43  ...    2.89    10.67   
2        의료용 물질 및 의약품 제조업   -2.09    16.09     77.05  ...   77.91   289.40   
3        의료용 물질 및 의약품 제조업    3.86    74.30     68.15  ...   37.80   132.06   
4        의료용 물질 및 의약품 제조업   -2.64    -5.37      1.66  ...   24.21    90.92   
...                   ...     ...      ...       ...  ...     ...      ...   
10056         도매 및 상품 중개업   -2.26    -2.56     -1.88  ...  102.82  3801.25   
10057         도매 및 상품 중개업    2.71    -7.36     37.82  ...  118.51  5704.66   
10058         도매 및 상품 중개업    2.47    -2.78     -3.98  ...  118.81  4046.12   
10059         도매 및 상품 중개업   -3.60    -0.50     10.54  ...   92.04  2603.45   
10060         도매 및 상품 중개업    1.77    11.95     25.00  ...   97.30  3777.94   

       부가가치율  노동소득분배율    자본분배율    이윤분배율  회계월  상장년도   회계년  연차  
0      19.56    89.82    10.18   -43.44   12  2001  2011  10  
1       1.66  1749.87 -1649.87 -2473.07   12  2001  2012  11  
2      34.29    71.28    28.72     4.23   12  2001  2013  12  
3      31.04    79.30    20.70     2.24   12  2001  2014  13  
4      15.96   144.10   -44.10   -84.58   12  2001  2015  14  
...      ...      ...      ...      ...  ...   ...   ...  ..  
10056   6.86    44.46    55.54    43.99   12  1994  2016  22  
10057   9.22    43.35    56.65    43.02   12  1994  2017  23  
10058   8.82    46.23    53.77    40.99   12  1994  2018  24  
10059   5.61    62.14    37.86    27.35   12  1994  2019  25  
10060   6.72    56.12    43.88    34.63   12  1994  2020  26  

[10061 rows x 59 columns]

In [25]:
df_con.to_csv("./데이터자료/2. 연결개별통합.csv",index=False)